spec/dummy/app/controllers/api/graphql_controller.rb

```diff
@@ -0,0 +1,15 @@
+require "manageiq/api/common/graphql"
+
+module Api
+  class GraphqlController < ApplicationController
+    def query
+      graphql_api_schema = ::ManageIQ::API::Common::GraphQL::Generator.init_schema(request)
+      variables = ::ManageIQ::API::Common::GraphQL.ensure_hash(params[:variables])
+      result = graphql_api_schema.execute(
+        params[:query],
+        :variables => variables
+      )
```

Comment:
It feels like we could encapsulate all of this into a simpler helper method such that this controller just becomes

```ruby
module Api
  class GraphqlController < ApplicationController
    def query
      render :json => ManageIQ::API::Common::GraphQL::ControllerAction.query(request)
    end
  end
end
```

and then separately

```ruby
module ManageIQ::API::Common::GraphQL::ControllerAction
  def self.query(request, params)
    GraphQL::Generator.init_schema(request).execute(
      params[:query],
      :variables => GraphQL.ensure_hash(params[:variables])
    )
  end
end
```

Reply:
Do you want this done here or in a separate PR ?

Reply:
Separate PR it is.